lib/elixir/lib/calendar/iso.ex

```diff
@@ -366,7 +366,7 @@ defmodule Calendar.ISO do
   @impl true
   def day_of_year(year, month, day)
       when is_integer(year) and is_integer(month) and is_integer(day) do
-    true = day <= days_in_month(year, month)
+    if day > days_in_month(year, month), do: raise(ArgumentError, "invalid date")
```

Comment:
Should we try to include the date in the error message?

Also, we should probably move this to a private function.

Reply:
We can emulate it with `invalid date: ~D[day: #{day}, month: #{month}, year: #{year}]`

Reply:
I think that mix of `~D` sigil and struct might be confusing, what do you think about?

```elixir
raise ArgumentError, "invalid date: #{date_to_string(year, month, day)}"
#=> "invalid date: 2000-12-53"
```